File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/Changes.java
package de.cubbossa.pathfinder;

import java.util.Collection;
import java.util.concurrent.ConcurrentHashMap;
import lombok.Getter;

@Getter
public final class Changes<E> {
  private final Collection<E> addList;
  private final Collection<E> removeList;

  public Changes() {
    addList = ConcurrentHashMap.newKeySet(16);
    removeList = ConcurrentHashMap.newKeySet(16);
  }

  public void flush() {
    addList.clear();
    removeList.clear();
  }
}

File: pathfinder-api/src/main/java/de/cubbossa/pathfinder/Changes.kt
package de.cubbossa.pathfinder

import java.util.concurrent.ConcurrentHashMap

class Changes<E> {
    val addList: MutableCollection<E> = ConcurrentHashMap.newKeySet(16)
    val removeList: MutableCollection<E> = ConcurrentHashMap.newKeySet(16)

    fun flush() {
        addList.clear()
        removeList.clear()
    }
}
